pyvista/core/filters/data_set.py

```diff
@@ -6854,11 +6860,20 @@ def color_labels(  # type: ignore[misc]
         >>> label_ids
         pyvista_ndarray([0, 1, 2, 3, 4])
 
-        Color the labels with the filter then plot them. Note that the
+        Color the labels with the filter. Note that the
         ``'glasbey_category10'`` color map is used by default.
 
-        >>> colored_labels = image_labels.color_labels()
-        >>> colored_labels.plot()
+        >>> colored_labels, color_dict = image_labels.color_labels(return_dict=True)
+
+        Plot the labels. We define a helper function to include a legend with the plot.
+
+        >>> def plot_labels(mesh, color_dict):
+        ...     pl = pv.Plotter()
+        ...     pl.add_mesh(mesh)
+        ...     pl.add_legend(list(color_dict.items()), face='triangle')
```

Comment:
This can be simplified with #7564 

```suggestion
        ...     pl.add_legend(color_dict)
```